lib/APPTurtle/src/Transports/CustomRosTransportTcp.h

```diff
@@ -0,0 +1,254 @@
+
+    /**
+     * Internal read wrapper with timeout handling loop.
+     * This function is used by the input state dependend read handlers.
+     *
+     * @param[out] buffer The buffer to read into.
+     * @param[in]  size The size of the buffer.
+     * @param[in]  timeout The timeout in milliseconds.
+     * @param[out] errorCode Set the error code to != 0 on error.
+     *
+     * @return The number of bytes read.
+     */
+    size_t readInternal(uint8_t* buffer, size_t size, int timeout, uint8_t* errorCode);
+
+    /**
+     * Try to read 2 byte size prefix (low, high) in InputState::INIT
+     * 
+     * @param[in]  timeout The timeout in milliseconds.
+     * @param[out] errorCode Set the error code to != 0 on error.
+     *
+     *  @return true if statemachine shall reloop.
+     */
+    bool readSizePrefix(int timeout, uint8_t* errorCode);
+
+    /**
+     * Try to read 2nd byte of size prefix (low, high) in InputState::PREFIX_1
+     * 
+     * Used in the rare event that low level read only provided one byte.
+     * 
+     * @param[in]  timeout The timeout in milliseconds.
+     * @param[out] errorCode Set the error code to != 0 on error.
+     *
+     * @return true if statemachine shall reloop.
+     */
+    bool readPendingSizePrefix(int timeout, uint8_t* errorCode);
+
+    /**
+     * Try payload bytes in InputState::PLAY_LOAD
+     * 
+     * Used in the rare event that low level read only provided one byte.
+     * 
+     * @param[in]  timeout The timeout in milliseconds.
+     * @param[out] errorCode Set the error code to != 0 on error.
+     * 
+     * @return true if statemachine shall reloop.
+     */
+    bool readPayload(int timeout, uint8_t* errorCode);
+
+    /**
+     * Handle message complete in InputState::FINISH
+     * 
+     * Used in the rare event that low level read only provided one byte.
+     * 
+     * @param[in]  timeout The timeout in milliseconds.
+     * @param[out] errorCode Set the error code to != 0 on error.
+     * 
+     * @return true if statemachine shall reloop.
+     */
+    bool readFinish(int timeout, uint8_t* errorCode);
+
+private:
+    WiFiClient            m_tcpClient;     /**< The TCP client.     */
+    static const String   m_protocolName;  /**< This protocol name. */
+    
+    /**
+     *  Read buffer handling from streaming sockets to implement framing.
+     *
+     * Custom transports with framing support need a customized agent.
+     * That is why the framing is implemented here to work with the
+     * "standard" tcpv4 agent protocol.   
+     */
+    enum InputState {
+        INIT,             /**< Input buffer ready for new record.          */
+        PREFIX_1,         /**< First byte of 2 -Byte size prefix received. */
+        PLAY_LOAD,        /**< Collecting payload bytes.                   */
+        FINISH,           /**< Payload record is complete.                 */
+        MAX               /**< Enum Range value, not a true state.         */
```

Comment:
Dangerous name for enumeration value.

According to path finder rule, the enum value prefix is "INPUT_STATE_"

Reply:
Done